### ISSUE TITLE
Request: Ability to send Wakeup Packet to Client

### ISSUE THREAD

**Issue by ghost** [input-leap/input-leap#1402]
> This issue has been migrated from old Barrier Github repository https://github.com/debauchee/barrier/issues/1402
> 
> Issue created on: 2021-11-05 by @shymega
> Issue last updated on: 2021-11-05

### Discussed in https://github.com/debauchee/barrier/discussions/1294

<div type='discussions-op-text'>

<sup>Originally posted by **ga2k** September 19, 2021</sup>
How hard would it be to implement the sending of a TCP/IP Wakeup Packet to a client so that we can use Barrier to login to a sleeping PC?

Ideally such a packet could be sent when the mouse tries to enter the client's activity area. Check if the client responds to a ping. If the client doesn't respond, send the wakeup packet and a message to the active screen saying "Waking up client - try again in a bit..."

This would have to be optional, of course, with a configuration option to enable/disable it.

Enhancements:

Maybe have a timer on the message with an option to cancel it, if the user was just waving the mouse around and didn't mean to go into the client's activation area. The message box say something like 

```
 ------------------------------------------------------
|                 Mac Mini is asleep.                 |
|                                                     |
|   We will try to wake up the client in 10 seconds   |
|         or you can click "Don't Switch".            |
|                                                     |
|    ------------                    --------------   |
|   | Do it now! |                  | Don't switch |  | 
|    ------------                    --------------   |
-------------------------------------------------------
```

 Then if the timer expires, send the wakeup packet, wait for the client to wake up, and perform the switch.

What do you think?</div>

**Comment by myn** [comment]
Would love WOL capabilities but perhaps passive available as an option "Wake client on activity" which would automatically send the WOL UDP packet to the client without having an active modal dialog prompt you to?

**Comment by CodingAlbert** [comment]
Absolutely, just passive available as an option is the way to go. Any activity regarding this by developers yet?
